_docs/persistenstorage-provider.md

```diff
@@ -0,0 +1,103 @@
+# Lets begin
+
+At Akash we use Rook/Ceph couple to provider dynamic volumes provisioning
+Ceph - distributed filesystem
+Rook - Kubernetes operator to control/provision Ceph
+
+In the guide **Rook** means Rook/Ceph couple.
+This guide is focused on Rook setup and usage for Akash provider. To get familiar with Rook concepts refer to the [official documentation](https://rook.github.io/docs/rook/v1.8/)
+Configuration is based on Rook V1.8
+
+## Prerequisites
+1. Helm charts referenced in the guide are present [here](https://github.com/ovrclk/helm-charts).
+2. Deploy and teardown are done by using bash script. Work on the helm chart is being done. Up until helm chart is ready this guide will be focused on creating test cluster only         
+Meanwhile examples referenced in the guide can be found [here](TBD)
+3. We will prevent cluster from using master nodes.
+4. We will be utilizing all available devices on each used node
+
+## Configuring
+In this example we are focusing on making slight adjustments to the `cluster.yaml` only. All other files must remain unchanged.
+
+1. get list of nodes in the cluster
+```shell
+kubectl get nodes -ojson | jq -r '.items[].metadata.labels."kubernetes.io/hostname"'
+k8s-master.edgenet-1.ewr1 # this node won't be used
+k8s-node-0.edgenet-1.ewr1
+k8s-node-1.edgenet-1.ewr1
+```
+
+2. replace `nodes:` in `cluster.yaml` with queried node above
+```yaml
+nodes:
+  - name: k8s-node-0.edgenet-1.ewr1
+  - name: k8s-node-1.edgenet-1.ewr1
+```
+
+## Deploying test cluster
+```shell
+# deploy takes around 10m
+ROOK_PATH=$(AKASH_ROOT)/_docs/rook/test ./scripts/rook.sh deploy 
+```
+
+## Check cluster health
+```shell
+ROOK_PATH=$(AKASH_ROOT)/_docs/rook/test ./scripts/rook.sh health
+```
+
+## Inventory operator
+
+Up until this point we were working on adjustments to the Kubernetes cluster itself
+
+1. Make sure akash provider is installed and running. Follow installation [guide](https://github.com/ovrclk/helm-charts#akash-provider-install) if not
+2. Install [inventory operator](https://github.com/ovrclk/helm-charts#akash-inventory-operator-optional---for-persistent-storage)
+
+## Teardown
+
+```shell
+ROOK_PATH=$(AKASH_ROOT)/_docs/rook/test ./scripts/rook.sh teardown
+```
+
+### Zapping Devices
+
+Disks on nodes used by Rook for osds can be reset to a usable state with the following method:
+
+Execute following script on each node that participated in rook cluster with list of comma separated devices needs to be wiped.
+For example devices `sda`,`sdc`, `sdg`
+```shell
+./zap.sh sda,sdc,sdg
+```
+
+#### zap.sh script
+```shell
+#!/usr/bin/env bash
+
+if [ "$#" -ne 1 ]; then
+	echo "Illegal number of parameters"
+fi
+
+# Zap the disk to a fresh, usable state (zap-all is important, b/c MBR has to be clean)
+IFS=","
+for disk in ${$1}; do
+	dev=/dev/${disk}
```

Comment:
This seems to be overly complex? As far as I can tell it 

1. Runs `sgdisk` which removes GPT & MBR
2. Wipes the first 100 MB of disk, which definitely removes GPT & MBR
3. OR it will run blkdiscard which wipes the entire disk

Can we just run one approach here? `dd` is highly reliable for this and available on basically every linux distribution for a long time.

Reply:
The dd approach will mark all sectors on the SSD as ‘used’ as it writes zeros across the disk. This interferes with the ability of the SSD to perform sector reallocations, native trim and wear levelling, as all sectors are now deemed occupied. blkdiscard on an SSD only marks the sectors as unused so that they are treated as all zeros

Reply:
While correct, if overwriting the first 100MB of a disk is an actual issue then we have a huge problem in production where the disk could be completely overwritten by one lease, then overwritten again by another lease, etc. I am sure lease owners will be writing more than 100MB at a time.

Can we not configure one of the pieces of storage management middleware (rook, ceph, whatever) to tell the hardware to perform the discard operation?